src/main/java/com/internet/shop/controllers/OrderDetailsController.java

```diff
@@ -0,0 +1,28 @@
+package com.internet.shop.controllers;
+
+import com.internet.shop.lib.Injector;
+import com.internet.shop.models.Product;
+import com.internet.shop.services.OrderService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class OrderDetailsController extends HttpServlet {
+    private static final Injector injector =
+            Injector.getInstance("com.internet.shop");
+    private OrderService orderService =
+            (OrderService) injector.getInstance(OrderService.class);
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String id = req.getParameter("id");
+        List<Product> products = orderService.get(Long.parseLong(id)).getProducts();
+        req.setAttribute("sum", orderService.getOrderPrice(Long.parseLong(id)));
```

Comment:
you already have order here with all products. There is no need to call a DB one more time inside getOrderPrice(). Let's pass the order to this method instead of id

Reply:
fixed